pkg/psalabelsyncer/podsecurity_label_sync_controller.go

```diff
@@ -301,11 +383,18 @@ func (c *PodSecurityAdmissionLabelSynchronizationController) checkNSControlled(n
 
 func checkNSControlled(ns *corev1.Namespace) bool {
 	nsName := ns.Name
-	isSystemNS := strings.HasPrefix(nsName, "openshift-") || nsName == "openshift" || nsName == "kube-system"
-	if isSystemNS {
+	if systemNSSyncExemptions.Has(nsName) {
 		return false
 	}
 
+	// while "openshift-" namespaces should be considered controller, there are some
+	// edge cases where users can also create them. Consider these a special case
+	// and delegate the decision to sync on the user who should know what they are
+	// doing when creating a NS that appears to be system-controlled.
+	if strings.HasPrefix(nsName, "openshift-") {
+		return ns.Labels[labelSyncControlLabel] == "true"
+	}
+
 	if ns.Labels[labelSyncControlLabel] != "false" {
```

Comment:
The current logic would consider a namespace `foo` labelled `security.openshift.io/scc.podSecurityLabelSync=meow` to be controlled but a namespace `openshift-foo` not.

Reply:
this is fine

Reply:
logically, yes, but the interpretation of `true`/`false` values is still inconsistent between the cases. the logic per se is fine, agreed.

Reply:
otoh, these are explicit values we're asking for here, so let's leave it.